xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-script/src/main/java/org/xwiki/user/script/GroupScriptService.java

```diff
@@ -140,4 +140,27 @@
     {
         return getMembers(group, true);
     }
+
+    /**
+     * Checks if a group (candidate) can be added to the members of another group (target).
+     * A group can be added to the members of another group if neither group already has the other in its members.
+     * A group cannot be added the its own members.
+     *
+     * @param target The targeted group
+     * @param candidate The group to be added
+     * @return {@code true} if the candidate can be added to the members of the target, {@code false} otherwise.
+     * @since 12.8RC1
+     */
+    public boolean canAddAsMember(DocumentReference target, DocumentReference candidate) throws GroupException
+    {
+        boolean ret;
+        if (target == null || candidate == null || candidate.equals(target)) {
+            ret = false;
+        } else if (getMembers(target, false).contains(candidate)) {
```

Comment:
Should be `true` here, you want all the members of that group, not just the direct ones.

Reply:
You you are right, I was too focused on the "make sure it's not already a member".